neurons/miner.py

```diff
@@ -398,6 +408,43 @@ def default_priority(self, synapse: bt.Synapse) -> float:
         )
         return priority
 
+    def get_config_for_test(self) -> bt.config:
+        return self.config
+
+    @property
+    def block(self):
+        return utils.ttl_get_block(self)
+
+    def sync(self):
+        """
+        Wrapper for synchronizing the state of the network for the given miner.
+        """
+        # Ensure miner hotkey is still registered on the network.
+        self.check_registered()
+
+        if self.should_sync_metagraph():
+            self.resync_metagraph()
+
+    def check_registered(self):
+        # --- Check for registration.
+        if not self.subtensor.is_hotkey_registered(
+            netuid=self.config.netuid,
+            hotkey_ss58=self.wallet.hotkey.ss58_address,
+        ):
+            bt.logging.error(
+                f"Wallet: {self.wallet} is not registered on netuid {self.config.netuid}."
+                f" Please register the hotkey using `btcli subnets register` before trying again."
+            )
+            sys.exit(1)
+
+    def should_sync_metagraph(self):
+        """
+        Check if enough epoch blocks have elapsed since the last checkpoint to sync.
+        """
+        return (
```

Comment:
Since we're no longer setting weights on the chain, this will always return true. Can we just switch to a time-based mechanism and update every 20 minutes or so?